validator_client/src/duties_service.rs

```diff
@@ -522,14 +543,29 @@ impl<T: SlotClock + 'static, E: EthSpec> DutiesService<T, E> {
         self.clone()
             .update_epoch(current_epoch + 1)
             .await
-            .map_err(move |e| {
+            .map_err(|e| {
                 error!(
                     log,
                     "Failed to get next epoch duties";
                     "http_error" => format!("{:?}", e)
                 );
             })?;
```

Comment:
@paulhauner To address your concern about block production not running when duties updates fail, what if we moved the channel broadcast (below) above this block that performs the update for the next epoch? Or we could change the next epoch update to be the same as the current epoch update above it (using `if let Err(e)` to log the error and keep running).

I'd prefer to do something like this than reintroduce the interval timer, because I think having the duties service act as the canonical source of time helps prevent inconsistency between the two services, and stops us from having to think about the block service running twice in every slot. With re-orgs, the block proposer duties could change mid-epoch, and we wouldn't want to produce a block using the old duties (on the interval) only to then get a duties update and realise we produced incorrectly.

Reply:
> what if we moved the channel broadcast (below) above this block that performs the update for the next epoch?

Yep, that solves my issue! Great alternative.